docs/design/architectural-overview.md

```diff
@@ -0,0 +1,168 @@
+
+
+
+### Control Plane Components and Responsibilities
+A control plane component is something responsible for accepting instruction via a CRD based API and ensuring that configuration is manifested into state that can be acted on.
+
+- **[Kuadrant Operator](https://github.com/Kuadrant/Kuadrant-operator):**
+    - Installation of data plane service protection components via their respective operators
+    - Exposes `RateLimitPolicy` and `AuthPolicy` and is currently the policy controller for these APIs
+    - Configures the Gateway to be able to leverage the data plane service protection components
+- **[Multi-Cluster Gateway Controller](https://github.com/Kuadrant/multicluster-gateway-controller):**
+    - Exposes `DNSPolicy` and `TLSPolicy`
+    - Configures DNS providers (e.g AWS Route 53) and TLS providers
+    - Focused around use cases involving distributed gateways (for example across clouds or geographic regions)
+    - Integrates with OCM as the multi-cluster management hub to distribute and observe gateway status based on the clusters they are deployed to works directly with OCM APIs such `PlacementDecision` and `ManifestWork`.
+- **[Kuadrant-add-on-manager](https://github.com/Kuadrant/multicluster-gateway-controller/cmd/ocm):**
+    - Sub component in the gateway controller repository
+    - Follows the [add-on pattern](https://open-cluster-management.io/concepts/addon/) from OCM
+    - Responsible for configuring and installing Kuadrant into a target spoke cluster
+
+- **[Limitador Operator:](https://github.com/Kuadrant/limitador-operator)**
+    - Installs and configures Limitador
+- **[Authorino Operator:](https://github.com/Kuadrant/authorino-operator)**
+    - Installs and configures Authorino
+
+### Data Plane Components and Responsibilities
+A data plane component sits in the request flow and is responsible for enforcing policy and providing service protection capabilities based on configuration managed and created by the control plane.
+
+- **Limitador:**
+    - Complies with the with Envoy rate limiting API to provide rate limiting to the gateway
+- **Authorino:**
+    - Complies with the Envoy external auth API to provide auth integration to the gateway
+- **WASM Plugin:**
+    - Uses the Proxy WASM ABI Spec to integrate with Envoy and provide filtering and connectivity to Authorino and Limitador for request time enforcement of auth and rate limiting
+
+
+### Dependencies and integrations
+
+In order to provide its full suite of functionality, Kuadrant has several dependencies. Some of these are optional depending on the functionality needed.
+
+- **[Cert-Manager](https://cert-manager.io/):** (provides the TLS integration)
+    - Optional: only required if you want to use TLSPolicy
```

Comment:
Likely we'll have to add something about being required for auth, but only after https://github.com/Kuadrant/kuadrant-operator/issues/207. And until https://github.com/Kuadrant/architecture/pull/16 (or any other option to install Kuadrant without auth), we may need to flag cert-manager we required.